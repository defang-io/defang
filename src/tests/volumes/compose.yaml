services:
  volume1:
    image: busybox
    volumes:
      - tmp:/tmp
  volume2:
    image: busybox
    volumes:
      - target: /tmp
        read_only: true
        type: volume

volumes:
  tmp:
