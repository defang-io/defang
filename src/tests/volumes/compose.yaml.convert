[
  {
    "name": "volume1",
    "image": "busybox",
    "platform": 2,
    "internal": true,
    "networks": 1,
    "volumes": [
      {
        "source": "tmp",
        "target": "/tmp"
      }
    ]
  },
  {
    "name": "volume2",
    "image": "busybox",
    "platform": 2,
    "internal": true,
    "networks": 1,
    "volumes": [
      {
        "target": "/tmp",
        "read_only": true
      }
    ]
  }
]