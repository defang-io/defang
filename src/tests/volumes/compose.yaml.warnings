level=warning msg="service \"volume1\": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)"
level=warning msg="service \"volume1\": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors"
level=warning msg="service \"volume2\": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)"
level=warning msg="service \"volume2\": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors"