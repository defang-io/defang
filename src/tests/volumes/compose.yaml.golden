name: volumes
services:
  volume1:
    image: busybox
    networks:
      default: null
    volumes:
      - type: volume
        source: tmp
        target: /tmp
        volume: {}
  volume2:
    image: busybox
    networks:
      default: null
    volumes:
      - type: volume
        target: /tmp
        read_only: true
networks:
  default:
    name: volumes_default
volumes:
  tmp:
    name: volumes_tmp
